benchbuild/project.py

```diff
@@ -227,6 +227,12 @@ def __check_if_uuid(self, _: tp.Any, value: uuid.UUID) -> None:  # pylint: disab
     source: Sources = attr.ib(
         default=attr.Factory(lambda self: type(self).SOURCE, takes_self=True))
 
+    primary_source: source.BaseSource = attr.ib(init=False)
+
+    @primary_source.default
+    def __default_primary_source(self) -> str:  # pylint: disable=no-self-use
+        return source.primary(*self.source).key
```

Comment:
Python complains to me that an instance of my project class (that inherits from `Project`) has no member with the name `source`. Not sure why, but using the `SOURCE` class attribute works.
I also just encountered a similar behavior with `Experiment.name`. Maybe it is a problem related to `attr`?

Reply:
hmmm not sure either, I experience weirdness with TypeErrors in the builder. Normally you can exploit
varargs to demand at least 1 function argument like I did with primary(...). Works fine in a python console  too, but yields TypeErrors during testing.

Reply:
Ah My Bad. That happends when the TestProject does not declare any source :-) Works as intended heh

Reply:
Seems to work now. Was the `init=False` the cause?

Reply:
Seems to work now. Was the `init=False` the cause?

Reply:
Not sure why it should change anything. Was a guess from my side. The attribute should be defined as soon as python loads the class.